controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -929,6 +973,14 @@ bool tlvPrefilledData::add_tlvs_from_list(std::list<wfa_ca::tlv_hex_t> &tlv_hex_
 
         auto values = string_utils::str_split(*tlv.value, ' ');
         for (auto value : values) {
+            //Do conversion if needed
+            if (net::network_utils::is_valid_mac(value)) {
+                mac_to_hex_notation(value);
```

Comment:
If the value is already hex, it is waste to do all these checks. Please do the check only if the value does not start with "0x".
Also, when thinking about it, there is no point of doing full validation again, because it has been validated before, so at this point, we know that the value is valid, so we only need to figure out which of them is the correct one.

```cpp
if (!value.find_first_of("x", 1, 1)) {
        if (value.find_first_of(":", 2, 1)) {
                mac_to_hex_notation(value);
        } else if  (value.find_first_of("b", 1, 1)) {
                binary_to_hex(value);
        } else if  (value.find_first_of("d", 1, 1)) {
                decimal_to_hex(value);
        }
}
```